WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java

```diff
@@ -177,12 +178,16 @@ public static void addMedia(Activity activity) {
     }
 
     public static void viewMyProfile(Context context) {
+        AccountHelper.getDefaultAccount().fetchAccountSettings();
+
         Intent intent = new Intent(context, MyProfileActivity.class);
         AnalyticsUtils.trackWithCurrentBlogDetails(AnalyticsTracker.Stat.OPENED_MY_PROFILE);
         slideInFromRight(context, intent);
     }
 
     public static void viewAccountSettings(Activity activity) {
+        AccountHelper.getDefaultAccount().fetchAccountSettings();
```

Comment:
IIRC, `ActivityLauncher` was introduced to keep the launch animations consistent for each activity (it also helps by killing some duplicate code). I don't have strong opinion on this point, it's a detail, choose your preferred solution ;). Are you listening for any events emitted by `fetchAccountSettings` in the activity, if yes, it makes sense to launch the call from the activity itself. 

Sidenote: We should also skip the call there if the network is down.